src/server.py

```diff
@@ -0,0 +1,29 @@
+"""server.py"""
+
+import socket
+import sys
+
+server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+server.bind(address)
+server.address(('127.0.0.1', 9999))
+server.listen(5)
+connection, address = server.accept()
+message = ''
+while True:
+while True:
+            buffer_length = 8
+            message_complete = False
+            while not message_complete:
+                part = connection.recv(buffer_length """BUFFER_SIZE""")
+                message += part.decode('utf8', 'ascii')
```

Comment:
So you're decoding each chunk as you get it. Since unicode characters can take up multiple bytes, what happens when a multiple-byte unicode character isn't received all the way in one chunk? i.e. the buffer length cuts it in half?

Rather than decoding one chunk at a time, what would be a better approach?

And anyway, for the purpose of this echo server, do you really need to decode the message at the server end, only to re-encode it and send it back?